arangod/Agency/Agent.cpp

```diff
@@ -1686,8 +1705,101 @@ void Agent::setPersistedState(VPackSlice const& compaction) {
   } catch (std::exception const& e) {
     LOG_TOPIC(ERR, Logger::AGENCY) << e.what() << " " << __FILE__ << __LINE__;
   }
+
 }
 
+bool Agent::haveJoinConfig() const {
+  MUTEX_LOCKER(guard, _joinLock);
+  return (_joinConfig != nullptr);
+}
+
+void Agent::join(query_t const& config) {
+
+  if (config != nullptr) {
+    _joinConfig = config;
```

Comment:
`_joinConfig` is not atomic, so this is a potential data race since it is outside the lock.